### ISSUE TITLE
Also generate ligatures for RTL emoji sequences 

### ISSUE THREAD

**Issue by anthrotype** [googlefonts/nanoemoji#365]
In https://github.com/googlefonts/nanoemoji/blob/main/src/nanoemoji/features.py, we only build GSUB ligatures for left-to-right (LTR) emoji sequences.

NotoColorEmoji.ttf also contains right-to-left (RTL) variants for those sequences that contain ZWJ (0x200D), see noto-emoji's `add_glyphs.py::get_rtl_seq` method for how to get the correct RTL sequence from the LTR one (it's not just reflect, the "Fitzpatrick modifiers" need to be treated specially): 
https://github.com/googlefonts/noto-emoji/blob/9a5261d871451f9b5183c93483cbd68ed916b1e9/add_glyphs.py#L227-L251

We should do the same here.

**Comment by khaledhosny** [comment]
Was this to work around https://github.com/harfbuzz/harfbuzz/issues/2265?

**Comment by anthrotype** [comment]
I don't think so, that issue refers to the directionality of region flags, these sequences are not for region flags, they are special emojis composed from other emojis https://blog.emojipedia.org/emoji-zwj-sequences-three-letters-many-possibilities/

**Comment by anthrotype** [comment]
cc @behdad @dougfelt

**Comment by anthrotype** [comment]
I found this in the android fontchain_lint.py script https://cs.android.com/android/platform/superproject/+/master:frameworks/base/tools/fonts/fontchain_linter.py;l=627

```python
        if _emoji_sequences.get(sequence, None) == 'Emoji_Tag_Sequence':
            # Add reverse of all emoji ZWJ sequences, which are added to the
            # fonts as a workaround to get the sequences work in RTL text.
            # TODO: test if these are actually needed by Minikin/HarfBuzz.
            reversed_seq = reverse_emoji(sequence)
            all_sequences.add(reversed_seq)
```

I don't know what's this about.

**Comment by anthrotype** [comment]
So, I just tested the sequence 👨‍💻 = 👨 + ZWJ + 💻 with these emoji fonts: NotoColorEmoji.ttf (original CBDT/CBLC bitmap color font, which contains additional reversed RTL ligatures for emoji ZWJ sequences), noto-glyf_colr_1.ttf (COLRv1 vector color font from https://github.com/googlefonts/color-fonts test repo, which doesn't have them) and AppleColorEmoji (from macOS 12, which similarly does not have them). I tried the canonical sequence in logical order, one wrapped with `<bdo dir="rtl">`, then one typed in reverse, and finally the reverse sequence with the RTL override.
Here is how they look in latest Chrome Canary 98:

<img width="508" alt="Screenshot 2021-12-22 at 16 52 03" src="https://user-images.githubusercontent.com/6939968/147127693-70dfff8d-f02a-470f-85da-9761f43fb6d2.png">

For all the three fonts, it appears that the ZWJ emoji sequences are invariant when written in logical order and the RTL direction override is enabled; only in the color bitmap NotoColorEmoji font, however, the sequences ligate and display the same even when typed in reverse logical order (with or without direction override).

I'm still unsure why NotoColorEmoji does that, and whether this belongs to a general purpose emoji compiler (like nanoemoji) or to Noto-specific post-processing.

**Comment by khaledhosny** [comment]
I'm not sure if bifi override case is something to be bothered with (e.g. `fi` with RTL bidi override should **not** form an fi ligature), I’d be more worried about emoji sequences in actual RTL text, so I’d repeat this test with emoji surrounded by two RTL characters instead of the bidi override.

**Comment by behdad** [comment]
Earlier versions of HarfBuzz didn't do anything special about emoji sequences. That meant that if these sequences appeared in out-of-context direction (eg. marked as Latin in right-to-left direction) then harfbuzz would flip them; that would make the GSUB rules not apply. Hence the reverse rules where added.

These days HB has special code to treat these sequences as atomic items, so when the buffer is being reversed they are reversed atomically. So that hack is not needed anymore.

**Comment by behdad** [comment]
> So that hack is not needed anymore.

This was fixed in https://github.com/harfbuzz/harfbuzz/issues/1159

Was released in HarfBuzz 2.0.0, October 2018.

**Comment by anthrotype** [comment]
Thanks Behdad. Then I think we can close the issue.
